onmt/data/Batch.lua

```diff
@@ -75,6 +75,8 @@ function Batch:__init(src, srcFeatures, tgt, tgtFeatures)
 
   self.sourceInputFeatures = {}
   self.sourceInputRevFeatures = {}
+  -- will be used to return extra padded value
+  self.padTensor = torch.LongTensor(self.size):fill(onmt.Constants.PAD)
```

Comment:
in pyramidal encoders - when calculating timestep t for level N, we will need to check value of time k*t, k*t+1, k*t+n. at the border, we will have sometime some overflow that is just handled with this automatic-padding